src/test/java/com.teragrep.jos_01.procfs/LinuxOSTest.java

```diff
@@ -0,0 +1,212 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs;
+
+import com.teragrep.jos_01.procfs.status.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class LinuxOSTest {
+
+    @Test
+    public void upTimeTest() {
+
+        LinuxOS os = new LinuxOS();
+        Uptime uptime = os.uptime();
+        Assertions.assertEquals(2, uptime.statistics().size());
+        Assertions.assertTrue(Float.parseFloat(uptime.statistics().get("uptimeSeconds")) > 0);
+        Assertions.assertTrue(Float.parseFloat(uptime.statistics().get("combinedCpuCoreIdleTimeSeconds")) > 0);
+    }
+
+    @Test
+    public void statTest() {
+
+        LinuxOS os = new LinuxOS();
+        OSStat stat = os.stat();
+        Assertions.assertEquals(12, stat.statistics().size());
+
+        String[] expectedKeys = {
+                "cpu",
+                "cpu0",
+                "cpu1",
+                "cpu2",
+                "cpu3",
+                "intr",
+                "ctxt",
+                "btime",
+                "processes",
+                "procs_running",
+                "procs_blocked",
+                "softirq"
+        };
+        for (String key : expectedKeys) {
+            Assertions.assertTrue(stat.statistics().containsKey(key));
+        }
+    }
+
+    @Test
+    public void cpuInfoTest() {
+        LinuxOS os = new LinuxOS();
+        Cpuinfo cpuinfo = os.cpuinfo();
+        Assertions.assertEquals(os.cpuThreadCount() * 27, cpuinfo.statistics().size());
```

Comment:
Where does this *27 come from?

Reply:
Each logical processor (or a cpu thread) should give 27 different pieces of information in \proc\cpuinfo, which should all be listed in cpuinfo.statistics. So there should be exactly 27*cpuThreadCount unique fields in cpuinfo.statistics

Reply:
Add that as a comment for future devs

Reply:
Added comment to test